soccer/Robot.cpp

```diff
@@ -522,64 +542,75 @@ void OurRobot::replanIfNeeded(const Geometry2d::CompositeShape& global_obstacles
 	full_obstacles.add(opp_obs);
 	full_obstacles.add(global_obstacles);
 
-	// if no goal command robot to stop in place
-	if (!_motionConstraints.targetPos) {
-		if (verbose) cout << "in OurRobot::replanIfNeeded() for robot [" << shell() << "]: stopped" << std::endl;
-		addText(QString("replan: no goal"), Qt::white, "Motion");
-
-		Planning::InterpolatedPath *newPath = new Planning::InterpolatedPath(pos);
-		setPath(newPath);
-		_path->draw(_state);
-		return;
-	}
-
-
-	Geometry2d::Point dest = *_motionConstraints.targetPos;
-
-	// //	if this number of microseconds passes since our last path plan, we automatically replan
-	const Time kPathExpirationInterval = 10 * SecsToTimestamp;
-	if ((timestamp() - _pathStartTime) > kPathExpirationInterval) {
-		_pathInvalidated = true;
-	}
-
-	if (!_path) {
-		_pathInvalidated = true;
-	} else {
-		_path->draw(_state, Qt::magenta);
-
-		//float maxDist = .6;
-		Point targetPathPos;
-		Point targetVel;
-		float timeIntoPath = ((float)(timestamp() - _pathStartTime)) * TimestampToSecs + 1.0f/60.0f;
-		_path->evaluate(timeIntoPath, targetPathPos, targetVel);
-		float pathError = (targetPathPos - pos).mag();
-		//state()->drawCircle(targetPathPos, maxDist, Qt::green, "MotionControl");
-		//addText(QString("velocity: %1 %2").arg(this->vel.x).arg(this->vel.y));
-		//addText(QString("%1").arg(pathError));
-		if (*_motionConstraints._replan_threshold!=0 && pathError > *_motionConstraints._replan_threshold) {
-			_pathInvalidated = true;
-			addText("pathError" , Qt::red, "Motion");
-			//addText(pathError);
-		}
-
-
-
-		if (_path->hit(full_obstacles, timeIntoPath)) {
-			_pathInvalidated = true;
-			addText("Hit Obstacle", Qt::red, "Motion");
-		}
-
-		//  invalidate path if current position is more than 15cm from the planned point
-		
-
-
-		//	if the destination of the current path is greater than X m away from the target destination,
-		//	we invalidate the path.  this situation could arise if during a previous planning, the target point
-		//	was blocked by an obstacle
-		//  TODO: This is Stupid. This should be fixed in the RRT planner or the Bezier Algorithm.
-		if (_path->destination() && (*_path->destination() - dest).mag() > 0.025) {
+	_state->drawCompositeShape(self_obs, Qt::gray, QString("self_obstacles_%1").arg(shell()));
+	_state->drawCompositeShape(opp_obs, Qt::gray, QString("opp_obstacles_%1").arg(shell()));
+	if (_path) {
```

Comment:
This whole thing is a mess. I'd prefer to fix this later in #377 as part of a bigger refactor.